src/foremast/awslambda/api_gateway_event/api_gateway_event.py

```diff
@@ -112,12 +112,41 @@ def add_permission(self):
         statement_id = '{}_api_{}'.format(self.app_name, self.trigger_settings['api_name'])
         principal = 'apigateway.amazonaws.com'
         lambda_alias_arn = get_lambda_alias_arn(self.app_name, self.env, self.region)
+        lambda_arn = get_lambda_arn(self.app_name, self.env, self.region)
+        resource_name = self.trigger_settings.get('resource', '')
+        resource_name = resource_name.replace("/", "")
+        source_arn = 'arn:aws:execute-api:{}:{}:{}/{}/{}/{}'.format(self.region, self.account_id, self.api_id, self.env, self.trigger_settings['method'], resource_name)
         add_lambda_permissions(function=lambda_alias_arn,
+                               statement_id=statement_id + self.trigger_settings['method'],
+                               action='lambda:InvokeFunction',
+                               principal=principal,
+                               env=self.env,
+                               region=self.region,
+                               source_arn=source_arn)
+        source_arn = 'arn:aws:execute-api:{}:{}:{}/*/*/{}'.format(self.region, self.account_id, self.api_id, resource_name)
+        add_lambda_permissions(function=lambda_alias_arn,
```

Comment:
The only difference between the first two `add_lambda_permissions()` calls is `function=lambda_alias_arn` and `function=lambda_arn`. Those values can be used in a `for` loop.